mood-analysis.rb

```diff
@@ -19,6 +20,55 @@ def analyze_mood(words)
   return ":-|"
 end
 
+def get_sanitized_input(words)
+  # input now matches feelings hash 
+  words.downcase!
+  # did not write a strip_punctuation method but this line strips it
+  words.gsub!(/[^a-z0-9\s]/i, '')
+  return words
+end 
+
+def get_happy_days(emoji_array)
+  happy = 0 
+  smile = ":-)"
+
+  number_of_days = 1 
+  emoji_array.each do |emoji|
+    if emoji == smile
+      happy += 1
+      if happy >= 3 && happy < 4
+        puts "It has taken #{number_of_days} days for #{happy} happy days to occur"
+      end
+    end
+    number_of_days += 1 
+  end
+  if happy > 3
+    puts "There have only been #{happy} happy days out of a total of #{number_of_days-1} days"
+  end 
+end
```

Comment:
This works, but I'd rather this method return a single fixnum opposed to the string with the fixnum in it. That leaves us more flexibility in using that information in a different way instead of being stuck to this specific use case.